nomad/structs/acl.go

```diff
@@ -345,6 +345,38 @@ func (a *ACLToken) HasRoles(roleIDs []string) bool {
 	return true
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface and allows
```

Comment:
The code here looks fine but in the other cases we have to support this kind of thing we do it at the `api` package layer. Ex. `api/acl.go` has this method for `ACLAuthMethod`. Should we do this parsing at the "boundary" in the API?

Reply:
Ah yes totally; I am not sure why I didn't use `ACLAuthMethod` as a basis considering I wrote that :|

I've force pushed the changes which includes `MarshalJSON` implementations and a slight change to account for parsing at the boundary which showed up during the bootstrap command. Thanks for spotting!